src/main/kotlin/app/batch/HBaseWriter.kt

```diff
@@ -18,46 +15,79 @@ import java.io.BufferedReader
 import java.io.InputStreamReader
 
 @Component
-class HBaseWriter(private val connection: Connection) : ItemWriter<DecompressedStream> {
+class HBaseWriter : ItemWriter<DecompressedStream> {
 
     @Autowired
     private lateinit var cipherService: CipherService
 
     @Autowired
     private lateinit var keyService: KeyService
 
+    @Autowired
+    private lateinit var hbase: HbaseClient
+
+    @Autowired
+    private lateinit var messageProducer: MessageProducer
+
+    @Autowired
+    private lateinit var messageUtils: MessageUtils
+
     override fun write(items: MutableList<out DecompressedStream>) {
         items.forEach {
+            logger.info("Processing '${it.fileName}'.")
             val fileName = it.fileName
-            val filenamePattern = """(?<database>[a-z-]+)\.(?<collection>[a-z-]+)\.\d+\.json\.gz\.enc$"""
+            val filenamePattern = """(?<database>[a-z0-9-]+)\.(?<collection>[a-z0-9-]+)\.\d+\.json\.gz\.enc$"""
```

Comment:
this could be a constant, and the compiled regex too?

Reply:
done - will appear after Uday next merges and pushes.